assignments/prototypes.js

```diff
@@ -41,70 +72,137 @@
-      width: 2,
-      height: 2,
+        length: 2,
+        width: 2,
+        height: 2,
     },
     healthPoints: 15,
     name: 'Sir Mustachio',
     team: 'The Round Table',
     weapons: [
-      'Giant Sword',
-      'Shield',
+        'Giant Sword',
+        'Giant Sword',
+        'Shield',
     ],
     language: 'Common Tongue',
-  });
+});
 
-  const archer = new Humanoid({
+const archer = new Humanoid({
     createdAt: new Date(),
     dimensions: {
-      length: 1,
-      width: 2,
-      height: 4,
+        length: 1,
+        width: 2,
+        height: 4,
     },
     healthPoints: 10,
     name: 'Lilith',
     team: 'Forest Kingdom',
     weapons: [
-      'Bow',
-      'Dagger',
+        'Bow',
+        'Dagger',
     ],
     language: 'Elvish',
-  });
-
-  console.log(mage.createdAt); // Today's date
-  console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
-  console.log(swordsman.healthPoints); // 15
-  console.log(mage.name); // Bruce
-  console.log(swordsman.team); // The Round Table
-  console.log(mage.weapons); // Staff of Shamalama
-  console.log(archer.language); // Elvish
-  console.log(archer.greet()); // Lilith offers a greeting in Elvish.
-  console.log(mage.takeDamage()); // Bruce took damage.
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.healthPoints); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.team); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(mage.takeDamage()); // Bruce took damage.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
+
+
+// Stretch task:
+// * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.
+// * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+function Villain(attributes)  {
+    Humanoid.call(this, attributes);
+    this.gasAttack = function(obj){
+        obj.healthPoints -= 15;
+        console.log(`${this.name} has attacked ${obj.name} with villainous laughing  gas! costs 15 healthpoints!`);
+        if(obj.healthPoints <= 0) {
+            console.log(obj.destroy());
+        }
+    }
+}
+
+Villain.protoype = Object.create(Humanoid.prototype);
+
+function Hero(attributes)  {
+    Humanoid.call(this, attributes);
+    this.ninjaChop = function(obj){
+        obj.healthPoints -= 5;
+        console.log(`${this.name} has attacked ${obj.name} with a heroic ninja chop! costs 5 healthpoints!`);
+        if(obj.healthPoints <= 0) {
```

Comment:
I can't tell why but, batman is not ninja-chopping the joker according to my developer console